Objects/longobject.c

```diff
@@ -359,9 +359,13 @@ PyLong_FromLong(long ival)
         if (IS_SMALL_UINT(ival)) { \
             return get_small_int((sdigit)(ival)); \
         } \
+        if ((ival) <= PyLong_MASK) { \
+            return _PyLong_FromMedium((sdigit)(ival)); \
+        } \
+        /* Do shift in two steps to avoid possible undefined behavior. */ \
+        INT_TYPE t = (ival) >> PyLong_SHIFT >> PyLong_SHIFT; \
         /* Count the number of Python digits. */ \
-        Py_ssize_t ndigits = 0; \
-        INT_TYPE t = (ival); \
+        Py_ssize_t ndigits = 2; \
```

Comment:
Depending on the `INT_TYPE` the value of ndigits can only be 2 or 3. We can replace the while loop with a single if statement. E.g. 

https://github.com/python/cpython/compare/main...eendebakpt:cpython:fast_digit_count

@srinivasreddy Would you be interested in benchmarking this approach and making a PR?

Reply:
Yes. Absolutely.

Reply:
Correct. We would need some guards for that. Whether that bit of complexity is worth is, depends on the benchmark results.